src/Vision/V1/ImageAnnotatorClient.php

```diff
@@ -30,13 +30,458 @@
+     * ```
+     *
+     * ```
+     * //[snippet=url]
+     * $imageUri = "gs://my-bucket/image.jpg";
+     * $image = $imageAnnotatorClient->createImageObject($imageUri);
+     * $response = $imageAnnotatorClient->faceDetection($image);
+     * ```
+     *
+     * @param  resource|string $imageInput An image to configure with
+     *         the given settings. This parameter will accept a resource, a
+     *         string of bytes, or the URI of an image in a publicly-accessible
+     *         web location.
+     * @return Image
+     * @throws InvalidArgumentException
+     */
+    public function createImageObject($imageInput)
+    {
+        return $this->createImageHelper(Image::class, ImageSource::class, $imageInput);
+    }
+
+    /**
+     * Run image detection and annotation for an image.
+     *
+     * Example:
+     * ```
+     * use Google\Cloud\Vision\V1\Feature;
+     * use Google\Cloud\Vision\V1\Feature_Type;
+     *
+     * $imageResource = fopen('path/to/image.jpg', 'r');
+     * $image = $imageAnnotatorClient->createImageObject($imageResource);
+     * $feature = new Feature();
+     * $feature->setType(Feature_Type::FACE_DETECTION);
+     * $features = [$feature];
+     * $response = $imageAnnotatorClient->annotateImage($image, $features);
+     * ```
+     *
+     * @param Image     $image        The image to be processed.
+     * @param Feature[] $features     Requested features.
+     * @param array     $optionalArgs {
+     *                                Optional.
+     *
+     *     @type ImageContext        $imageContext  Additional context that may accompany the image.
+     *     @type RetrySettings|array $retrySettings
+     *          Retry settings to use for this call. Can be a
+     *          {@see Google\ApiCore\RetrySettings} object, or an associative array
+     *          of retry settings parameters. See the documentation on
+     *          {@see Google\ApiCore\RetrySettings} for example usage.
+     * }
+     *
+     * @return AnnotateImageResponse
+     *
+     * @throws ApiException if the remote call fails
+     * @experimental
+     */
+    public function annotateImage($image, $features, $optionalArgs = [])
+    {
+        return $this->annotateImageHelper(
+            [$this, 'batchAnnotateImages'],
+            AnnotateImageRequest::class,
+            $image,
+            $features,
+            $optionalArgs
+        );
+    }
+
+    /**
+     * Run face detection for an image.
+     *
+     * Example:
+     * ```
+     * $imageContent = file_get_contents('path/to/image.jpg');
+     * $image = $imageAnnotatorClient->createImageObject($imageContent);
+     * $response = $imageAnnotatorClient->faceDetection($image);
+     * ```
+     *
+     * @param Image $image        An image annotation request.
+     * @param array $optionalArgs {
+     *                            Optional.
+     *
+     *     @type RetrySettings|array $retrySettings
```

Comment:
Does `ImageContext` also apply as an optional argument for the one off helpers?

Reply:
Yes it does - i will add it

Reply:
Done